### PR TITLE
Add search filters and pagination for rooms

### PR DESCRIPTION
**Pull request by leonbsilva**

### CODE REVIEW

File: lib/docusign_dtr/query_param_helper.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      def search
+        @options[:search].to_s
+      end
+
+      def count
+        @options[:count].to_i
+      end
+
+      def start_position
+        # binding.pry
```
Comment (camilosad) [inline lib/docusign_dtr/query_param_helper.rb]: ✂️

File: lib/docusign_dtr/query_param_helper.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        @options[:startPosition].to_i
+      end
+
+      def room_status
+        raise "error: not acceptable value #{@options[:room_status]}" unless %w[
```
Comment (camilosad) [inline lib/docusign_dtr/query_param_helper.rb]: do you think it makes sense to move this list of allowed options to the model instead of hardcoding it here?

File: lib/docusign_dtr/query_param_helper.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      sort: :sort,
+      dateRangeType: :date_range_type,
+      startDate: :start_date
+    }.freeze
+
+    class << self
+      def call(options)
+        @options = options
+        QUERY_PARAMS.each_with_object({}) do |(key, value), memo|
+          memo[key] = send(value) if @options.key? key
```
Comment (sdavies) [inline lib/docusign_dtr/query_param_helper.rb]: I think this might be backwards.  

When we call this from the ruby side we should use ```{has_contract_amount: true}``` NOT ```{hasContractAmount: true}```

File: lib/docusign_dtr/query_param_helper.rb [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      def has_submitted_task_list
+        to_boolean(@options[:has_submitted_task_list])
+      end
+
+      def has_contract_amount
+        to_boolean(@options[:has_contract_amount])
+      end
+
+      def sort
+        raise "error: not acceptable value #{@options[:sort]}" unless acceptable_value?(:sort, @options[:sort].to_s)
```
Comment (sdavies) [inline lib/docusign_dtr/query_param_helper.rb]: can we define a custom error here?
class InvalidParameter < StandardError; end
raise InvalidParameter.new("value #{value} is not valid for #{key}") or something along those lines?

This could be in a separate PR if you wish!

Comment (leonbsilva) [reply lib/docusign_dtr/query_param_helper.rb]: I will refactor and cover with tests in a new PR!
